### DISCUSSION TITLE
How to customize list item indentation

### DISCUSSION THREAD

**Discussion by ivan-lednev** [syntax-tree/unist#45] [Q&A] [answered]
I'd like to be able to set the number of spaces before & after each bullet in a list-item. Currently, it looks like I'm stuck with either of these two options:
```
Two spaces before, one space after:

- One
  - Two

Four spaces before, three after:

-   One
    -   Two
```

Is it possible to somehow extend the list-item handler? Or copying & modifying this project locally is my only option? Many thanks.

**Comment by wooorm** [comment, marked as answer]
Can you provide more info? See https://github.com/syntax-tree/.github/blob/main/support.md#asking-quality-questions.

**Reply by ivan-lednev** [reply]
Hello, 

there is a plugin for a markdown editor, that parses files and then uses [mdast-util-to-markdown](https://github.com/syntax-tree/mdast-util-to-markdown) to prettify file contents.

In mdast-util-to-markdown, the configuration provides only 2 ways of formatting lists:
[source code](https://github.com/syntax-tree/mdast-util-to-markdown/blob/f3df7410049ed426ef8734ec762a38aa2feee73f/lib/util/check-list-item-indent.js#L10)

"one", which means two spaces before, one space after each bullet:

```
- One
  - Two
```

"tabs", which means four spaces before, three spaces after each bullet:

```
-   One
    -   Two
```

However, I'd like to make the formatter more configurable, to be able to do, say, 4 spaces before, one space after each bullet:

```
- One
    - Two
```

I looked around the code and saw that the code I'm interested in is here: https://github.com/syntax-tree/mdast-util-to-markdown/blob/main/lib/handle/list-item.js

However, it doesn't seem like the customization I'm interested in is possible. Therefore, my question is: what is the best way to achieve this kind of customization?

**Reply by wooorm** [reply]
Many markdown parsers are notoriously bad at list indents. I strongly suggest not to do your “expected” format and stick with 1 space or tab stop instead

**Reply by wooorm** [reply]
This also conflicts with how CommonMark handles lists and code: you’ll run into problems there too

**Reply by ivan-lednev** [reply]
Ok, thanks a lot for a quick response!
